addons/test_mail/tests/test_discuss.py

```diff
@@ -130,3 +130,17 @@ def test_mail_cc_recipient_suggestion(self):
             (False, 'cc1@example.com', 'CC Email'),
             (False, 'cc2@example.com', 'CC Email'),
         ], 'cc should be in suggestions')
+
+    def test_unlink_notification_message(self):
+        channel = self.env['mail.channel'].with_user(self.user_employee).create({'name': 'testChannel'})
+
+        channel.message_notify(
+            body='test',
+            message_type='user_notification',
+            partner_ids=[self.partner_2.id],
+            author_id=2
+        )
+        channel.with_user(self.user_employee).unlink()
+
+        remaining_messages = self.env['mail.message'].search([('model', '=', 'mail.channel'), ('res_id', 'in', channel.ids)])
```

Comment:
Technically you could check messages before the unlink (should have the test + maybe creation one ?), then do an exists post-unlink and check those have been removed :) .

Reply:
Done :+1: